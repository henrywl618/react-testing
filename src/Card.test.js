import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Card from "./Card";

it("renders without crashing", function() {
    render(<Card src="./image1.jpg" caption="Photo by Richard Pasquarella on Unsplash" currNum={1} totalNum={3}/>)
});

it("matches the snapshot", function() {
    const {asFragment} = render(<Card src="./image1.jpg" caption="Photo by Richard Pasquarella on Unsplash" currNum={1} totalNum={3}/>);
    expect(asFragment()).toMatchSnapshot();
});

it("matches the snapshot", function() {
    const {asFragment} = render(<Card src="./image2.jpg" caption="Photo by Pratik Patel on Unsplash" currNum={2} totalNum={3}/>);
    expect(asFragment()).toMatchSnapshot();
});
