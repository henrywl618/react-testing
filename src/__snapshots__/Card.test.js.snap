// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="./image1.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 3.
    </small>
  </div>
</DocumentFragment>
`;

exports[`matches the snapshot 2`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Pratik Patel on Unsplash
    </h4>
    <img
      alt="Photo by Pratik Patel on Unsplash"
      class="Card-image"
      src="./image2.jpg"
    />
    <small
      class="Card-small"
    >
      Image 2 of 3.
    </small>
  </div>
</DocumentFragment>
`;
